pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
+	// Who currently has the lock.
+	HeldBy string `json:"heldby"`
+	// How long the lock is valid.
+	LeaseTime uint64 `json:"duration"`
+	// When the lock was first acquired.
+	AcquiredTime string `json:"atime,omitempty"`
+	// When the lock was last renewed.
+	RenewTime string `json:"rtime,omitempty"`
```

Comment:
@mikedanese 
"LeaseTime, AcquireTime, RenewTime are not sufficient to guarantee to a lease holder proper exclusive rights to resource access. To guarantee serialization of resource access the lease holder needs to acquire a lease session identifier or a sequence number (e.g. resource version in kubernetes or index in etcd) and pass it with mutating rpcs so that the lease can be checked during the commit. Exposing these times as actionable datapoints will lead to missuse of the API. Is this avoidable using the etcd v2 api? We use the lease transaction when we move to v3."

Yes, this is mentioned in the design documentation.  There will be a small window of opportunity for split brain that can be closed up with the newer etcd api.  This issues has been accepted as is in the proposal with the intention of closing the loop once etcd supports the needed functionality.